server/src/main/java/org/eclipse/openvsx/adapter/VSCodeAPI.java

```diff
@@ -99,6 +141,7 @@ public ExtensionQueryResult extensionQuery(@RequestBody ExtensionQueryParam para
             }
         }
 
+        // TODO can be replaced with LocalVSCodeService.toQueryResult
```

Comment:
no, it is just mention that this file has code duplication, and rely on how LocalVSCodeService compose response, it would be good to have one function which produce consistent result for both

Reply:
Maybe move `toQueryResult` to a util class (then you can use `static`) or move it to `VSCodeAPI` since both `local` and `upstream` use it.